### DISCUSSION TITLE
sklearn feature selection and cross validation for multivariate variable MiniROCKET

### DISCUSSION THREAD

**Discussion by riedejon** [alan-turing-institute/sktime#4229] [Q&A]
Hello fellow programmers

I am currently working on a project that involves time series classification. I am using MniROCKET since it is super fast and accurate. To be specific I use the MiniRocketMultivariateVariable. Is it possible to run sequential feature selection (https://scikit-learn.org/stable/modules/feature_selection.html) and the cross-validation (https://scikit-learn.org/stable/modules/cross_validation.html)?

If yes can you help me with the implementation?

Thank you very much.

Best regards,
Jonathan

**Comment by fkiraly** [comment]
Hm, I see - what you are looking for, is it this algorithm?

1. first apply MiniROCKET to get features
2. then apply some feature selection algo from sklearn to subselect them
3. then apply an sklearn classifier

Yes?

To do this, the following code should do the trick (first, without grid search)

```python
from sktime.transformations.panel.rocket import MiniRocketMultivariateVariable
from sklearn.feature_selection import SelectKBest
from sklearn.ensemble import RandomForestClassifier
from sklearn.pipeline import make_pipeline

rocket = MiniRocketMultivariateVariable()
fselect = SelectKBest()
sklearn_clf = RandomForestClassifier()
rocket_select_clf = rocket * make_pipeline(fselect, sklearn_clf)
```

Here, `rocket_select_clf` will work as any `sktime` classifier.

The trick is pipelining the feature selection and sklearn classifier using the sklearn pipeline.

For grid search, you can use `sklearn` `GridSearchCV` on top of this - note that this will only work for `sktime` data containers where `shape[0]` = number instances.
We are currently working on an `sktime` grid search for classification that allows more data types, see https://github.com/sktime/sktime/issues/4129 (FYI @achieveordie)

**Reply by riedejon** [reply]
Hello @fkiraly thank you for answering so quickly. Exactly. My idea was first to transform my X_data and then apply the feature classification with the classifier and the transformed data. I tried something like this X_transform = MiniROCKET.transform(X) but the output did not seem right to me.

I will definitely try your pipeline approach. After I created the pipeline with the feature selection do I fit it on the whole X_data?
